From 6b27c0f334398076ef403cc604420df351d8f36e Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 09 Nov 2022 10:03:30 +0200
Subject: [PATCH] [SW-81037] gaudi3: fix write inflight polling

Wrong register was used for monitoring writes inflight,
this is fixed by using the dedicated macro for polling
on writes inflight.

Change-Id: Icf5a39c6628ea88c3f25d1af85ce6da6a4920a81
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 3aaeac2..c3017c4 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -586,8 +586,7 @@
 	 */
 	memcpy((void *)mem_addr, (void *)buff_addr, PIF_ARC_CCM_SIZE);
 	/* wait for all writes to complete */
-	while (!regmap_readl(mmD0_PIF_ARC_CONF_NO_WR_INFLIGHT))
-		;
+	HL_ARC_W_BARRIER();
 
 	regmap_writel(CCM_TERMINATE_MASK, ccm_terminate_reg);
 }
